File: src/transformers/features/http/CreateHttpParameterTransformer.ts
import { HttpParameterProgrammer } from "../../../programmers/http/HttpParameterProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpParameterTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createParameter",
      write: HttpParameterProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/CreateHttpParameterTransformer.go
package http

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type createHttpParameterTransformerNamespace struct{}

var CreateHttpParameterTransformer = createHttpParameterTransformerNamespace{}

func (createHttpParameterTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "http.createParameter",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativehttpprogrammers.HttpParameterProgrammer.Write(x)
    },
  })
}
